### ISSUE TITLE
lme4 models with two random effects cannot be displayed

### ISSUE THREAD

**Issue by elbersb** [vincentarelbundock/modelsummary#288]
With two random effects, `parameters` (and also `broom.mixed::tidy`) returns two variables `SD (Intercept)`, which then trips up `modelsummary`. See here:

``` r
library("lme4")
#> Loading required package: Matrix
library("modelsummary")

mod <- lmer(mpg ~ (1|am) + (1|cyl), data = mtcars)
msummary(mod, output = "markdown")
#> Error in map_omit_rename_estimates(tmp, coef_rename = coef_rename, coef_map = coef_map, : Two coefficients from a single model cannot share the same name.


(par <- parameters::parameters(mod, effects = "all"))
#> # Fixed Effects
#> 
#> Parameter   | Coefficient |   SE |         95% CI | t(28) |      p
#> ------------------------------------------------------------------
#> (Intercept) |       20.64 | 3.26 | [14.25, 27.03] |  6.33 | < .001
#> 
#> # Random Effects
#> 
#> Parameter           | Coefficient
#> ---------------------------------
#> SD (Intercept: cyl) |        5.15
#> SD (Intercept: am)  |        1.71
#> SD (Residual)       |        1.75
parameters::standardize_names(par, style = "broom")
#>                term  estimate std.error conf.level conf.low conf.high statistic
#> 1       (Intercept) 20.638942  3.260157       0.95 14.24915  27.02873   6.33066
#> 2    SD (Intercept)  5.150924        NA       0.95       NA        NA        NA
#> 3    SD (Intercept)  1.709297        NA       0.95       NA        NA        NA
#> 4 SD (Observations)  1.751055        NA       0.95       NA        NA        NA
#>   df.error      p.value effect    group
#> 1       28 2.441155e-10  fixed         
#> 2       NA           NA random      cyl
#> 3       NA           NA random       am
#> 4       NA           NA random Residual
```
<sup>Created on 2021-05-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

For an ideal `lmer` output, one would want the `effect` and `group` variables to be displayed in the table, but that may require a solution that's handcrafted for `lme4`. I'm not sure how many models return additional columns beyond term, estimate, etc., but possibly one solution would be to check for repeated variable names in the object returned from broom/parameters, and *if* there are duplicates, then to merge any additional character columns into the `term` column. This should be fairly simple to implement, but may have all kinds of side effects...

**Comment by vincentarelbundock** [comment]
Thanks! I fixed the bug in d23cc7b

The proper way to handle models like this is to use the `group` argument. After the bugfix, you get an informative warning and the intended behavior works as expected:

``` r
library("lme4")
#> Loading required package: Matrix
library("modelsummary")

mod <- lmer(mpg ~ hp + (1 | am) + (1 | cyl), data = mtcars)

msummary(mod, output = "markdown")
#> Warning in msummary(mod, output = "markdown"): The table includes duplicate term
#> names. This can sometimes happen when a model produces "grouped" terms, such as
#> in multinomial logit or gamlss models. You may want to call `get_estimates(model)`
#> to see how estimates are labelled internally, and use the `group` argument of the
#> `modelsummary` function.
```

<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">Model 1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: center;">27.827</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: center;">(3.024)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">hp</td>
<td style="text-align: center;">-0.051</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: center;">(0.012)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SD (Intercept)</td>
<td style="text-align: center;">1.874</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: center;">3.175</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SD (Observations)</td>
<td style="text-align: center;">1.640</td>
</tr>
<tr class="even">
<td style="text-align: left;">Num.Obs.</td>
<td style="text-align: center;">32</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R2 Marg.</td>
<td style="text-align: center;">0.367</td>
</tr>
<tr class="even">
<td style="text-align: left;">R2 Cond.</td>
<td style="text-align: center;">0.780</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AIC</td>
<td style="text-align: center;">175.3</td>
</tr>
<tr class="even">
<td style="text-align: left;">BIC</td>
<td style="text-align: center;">182.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ICC</td>
<td style="text-align: center;">0.7</td>
</tr>
<tr class="even">
<td style="text-align: left;">RMSE</td>
<td style="text-align: center;">2.50</td>
</tr>
</tbody>
</table>

``` r
msummary(mod, group = group + term ~ model, output = "markdown")
```

<table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;"></th>
<th style="text-align: center;">Model 1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: center;">27.827</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: center;">(3.024)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">hp</td>
<td style="text-align: center;">-0.051</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: center;">(0.012)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cyl</td>
<td style="text-align: left;">SD (Intercept)</td>
<td style="text-align: center;">1.874</td>
</tr>
<tr class="even">
<td style="text-align: left;">am</td>
<td style="text-align: left;"></td>
<td style="text-align: center;">3.175</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Residual</td>
<td style="text-align: left;">SD (Observations)</td>
<td style="text-align: center;">1.640</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">Num.Obs.</td>
<td style="text-align: center;">32</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">R2 Marg.</td>
<td style="text-align: center;">0.367</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">R2 Cond.</td>
<td style="text-align: center;">0.780</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">AIC</td>
<td style="text-align: center;">175.3</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">BIC</td>
<td style="text-align: center;">182.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">ICC</td>
<td style="text-align: center;">0.7</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">RMSE</td>
<td style="text-align: center;">2.50</td>
</tr>
</tbody>
</table>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.3 (2020-10-10)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_CA.UTF-8                 
#>  ctype    en_CA.UTF-8                 
#>  tz       America/Toronto             
#>  date     2021-05-10                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package        * version    date       lib
#>  assertthat       0.2.1      2019-03-21 [1]
#>  backports        1.2.1      2020-12-09 [1]
#>  bayestestR       0.9.0      2021-04-08 [1]
#>  boot             1.3-25     2020-04-26 [2]
#>  broom            0.7.6      2021-04-05 [1]
#>  checkmate        2.0.0      2020-02-06 [1]
#>  cli              2.5.0      2021-04-26 [1]
#>  colorspace       2.0-1      2021-05-04 [1]
#>  crayon           1.4.1      2021-02-08 [1]
#>  DBI              1.1.1      2021-01-15 [1]
#>  digest           0.6.27     2020-10-24 [1]
#>  dplyr            1.0.6      2021-05-05 [1]
#>  ellipsis         0.3.2      2021-04-29 [1]
#>  evaluate         0.14       2019-05-28 [1]
#>  fansi            0.4.2      2021-01-15 [1]
#>  fs               1.5.0      2020-07-31 [1]
#>  generics         0.1.0      2020-10-31 [1]
#>  glue             1.4.2      2020-08-27 [1]
#>  highr            0.9        2021-04-16 [1]
#>  htmltools        0.5.1.1    2021-01-22 [1]
#>  httr             1.4.2      2020-07-20 [1]
#>  insight          0.14.0     2021-05-10 [1]
#>  kableExtra       1.3.4.9000 2021-04-04 [1]
#>  knitr            1.33       2021-04-24 [1]
#>  languageserver * 0.3.9      2020-12-18 [1]
#>  lattice          0.20-41    2020-04-02 [1]
#>  lifecycle        1.0.0      2021-02-15 [1]
#>  lme4           * 1.1-26     2020-12-01 [1]
#>  magrittr         2.0.1      2020-11-17 [1]
#>  MASS             7.3-53.1   2021-02-12 [1]
#>  Matrix         * 1.3-2      2021-01-06 [1]
#>  minqa            1.2.4      2014-10-09 [1]
#>  modelsummary   * 0.7.0.9000 2021-05-10 [1]
#>  munsell          0.5.0      2018-06-12 [1]
#>  nlme             3.1-152    2021-02-04 [1]
#>  nloptr           1.2.2.2    2020-07-02 [1]
#>  parameters       0.13.0.1   2021-05-10 [1]
#>  performance      0.7.1      2021-04-09 [1]
#>  pillar           1.6.0      2021-04-13 [1]
#>  pkgconfig        2.0.3      2019-09-22 [1]
#>  purrr            0.3.4      2020-04-17 [1]
#>  R.cache          0.14.0     2019-12-06 [1]
#>  R.methodsS3      1.8.1      2020-08-26 [1]
#>  R.oo             1.24.0     2020-08-26 [1]
#>  R.utils          2.10.1     2020-08-26 [1]
#>  R6               2.5.0      2020-10-28 [1]
#>  Rcpp             1.0.6      2021-01-15 [1]
#>  reprex           1.0.0      2021-01-27 [1]
#>  rlang            0.4.11     2021-04-30 [1]
#>  rmarkdown        2.8        2021-05-07 [1]
#>  rstudioapi       0.13       2020-11-12 [1]
#>  rvest            1.0.0      2021-03-09 [1]
#>  scales           1.1.1      2020-05-11 [1]
#>  sessioninfo      1.1.1      2018-11-05 [1]
#>  statmod          1.4.35     2020-10-19 [1]
#>  stringi          1.6.1      2021-05-10 [1]
#>  stringr          1.4.0      2019-02-10 [1]
#>  styler           1.3.2      2020-02-23 [1]
#>  svglite          2.0.0      2021-02-20 [1]
#>  systemfonts      1.0.1      2021-02-09 [1]
#>  tables           0.9.6      2020-09-22 [1]
#>  tibble           3.1.1      2021-04-18 [1]
#>  tidyr            1.1.3      2021-03-03 [1]
#>  tidyselect       1.1.1      2021-04-30 [1]
#>  utf8             1.2.1      2021-03-12 [1]
#>  vctrs            0.3.8      2021-04-29 [1]
#>  viridisLite      0.4.0      2021-04-13 [1]
#>  webshot          0.5.2      2019-11-22 [1]
#>  withr            2.4.1      2021-01-26 [1]
#>  xfun             0.22       2021-03-11 [1]
#>  xml2             1.3.2      2020-04-23 [1]
#>  yaml             2.2.1      2020-02-01 [1]
#>  source                               
#>  standard (@0.2.1)                    
#>  standard (@1.2.1)                    
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.3)                       
#>  standard (@1.4.1)                    
#>  standard (@1.1.1)                    
#>  standard (@0.6.27)                   
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.3)                       
#>  standard (@0.14)                     
#>  standard (@0.4.2)                    
#>  standard (@1.5.0)                    
#>  CRAN (R 4.0.2)                       
#>  standard (@1.4.2)                    
#>  CRAN (R 4.0.2)                       
#>  standard (@0.5.1.1)                  
#>  standard (@1.4.2)                    
#>  Github (easystats/insight@69f5cdf)   
#>  Github (haozhu233/kableExtra@a6af5c0)
#>  CRAN (R 4.0.3)                       
#>  standard (@0.3.9)                    
#>  CRAN (R 4.0.3)                       
#>  standard (@1.0.0)                    
#>  standard (@1.1-26)                   
#>  standard (@2.0.1)                    
#>  standard (@7.3-53.)                  
#>  CRAN (R 4.0.2)                       
#>  standard (@1.2.4)                    
#>  local                                
#>  standard (@0.5.0)                    
#>  CRAN (R 4.0.2)                       
#>  standard (@1.2.2.2)                  
#>  Github (easystats/parameters@411f85f)
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.2)                       
#>  standard (@2.0.3)                    
#>  standard (@0.3.4)                    
#>  standard (@0.14.0)                   
#>  standard (@1.8.1)                    
#>  standard (@1.24.0)                   
#>  standard (@2.10.1)                   
#>  standard (@2.5.0)                    
#>  CRAN (R 4.0.2)                       
#>  standard (@1.0.0)                    
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.3)                       
#>  standard (@0.13)                     
#>  CRAN (R 4.0.3)                       
#>  standard (@1.1.1)                    
#>  standard (@1.1.1)                    
#>  standard (@1.4.35)                   
#>  CRAN (R 4.0.3)                       
#>  standard (@1.4.0)                    
#>  standard (@1.3.2)                    
#>  CRAN (R 4.0.3)                       
#>  standard (@1.0.1)                    
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.2)                       
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.3)                       
#>  CRAN (R 4.0.2)                       
#>  standard (@0.5.2)                    
#>  standard (@2.4.1)                    
#>  CRAN (R 4.0.3)                       
#>  standard (@1.3.2)                    
#>  standard (@2.2.1)                    
#> 
#> [1] /Users/vincent/Library/R/4.0/library
#> [2] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

**Comment by elbersb** [comment]
Oops, should've read the docs! But the warning is great!
